insteon_mqtt/handler/StandardCmdNAK.py

```diff
@@ -0,0 +1,72 @@
+#===========================================================================
+#
+# Insteon broadcast message handler
+#
+#===========================================================================
+from .. import log
+from .. import message as Msg
+from .StandardCmd import StandardCmd
+
+LOG = log.get_logger()
+
+
+class StandardCmdNAK(StandardCmd):
+    """Insteon standard input mesage handler that passes on NAKs
+
+    The Standard command handler will process and not pass on Naks to the
+    callback function.
+
+    This is handler is identical in everyway, but will pass on a NAK so that
+    the callback can process it.  This is rarely needed
+    """
+
+    #-----------------------------------------------------------------------
+    def msg_received(self, protocol, msg):
+        """See if we can handle the message.
+
+        See if the message is the expected ACK of our output or the expected
+        InpStandard reply message.  If we get a reply, pass it to the
+        callback.
+
+        Args:
+          protocol (Protocol):  The Insteon Protocol object
+          msg: Insteon message object that was read.
+
+        Returns:
+          Msg.UNKNOWN if we can't handle this message.
+          Msg.CONTINUE if we handled the message and expect more.
+          Msg.FINISHED if we handled the message and are done.
+        """
+        # Probably an echo back of our sent message.
+        if isinstance(msg, Msg.OutStandard):
+            # If the message is the echo back of our message, then continue
+            # waiting for a reply.
+            if msg.to_addr == self.addr and msg.cmd1 == self.cmd:
+                if not msg.is_ack:
+                    LOG.error("%s NAK response", self.addr)
+
+                LOG.debug("%s got msg ACK", self.addr)
+                return Msg.CONTINUE
+
+            # Message didn't match the expected addr/cmd.
+            LOG.debug("%s handler unknown msg", self.addr)
+            return Msg.UNKNOWN
+
+        # See if this is the standard message ack/nak we're expecting.
+        elif isinstance(msg, Msg.InpStandard):
+            # If this message matches our address and command, it's probably
+            # the ACK we're expecting.
+            if msg.from_addr == self.addr and msg.cmd1 == self.cmd:
+                # Run the callback it decides what to do with an ACK or NAK
+                self.callback(msg, on_done=self.on_done)
+
+                # Indicate no more messages are expected.
+                return Msg.FINISHED
+            else:
```

Comment:
Do you still want the "unexpected message" printout for messages that don't match from_addr and cmd1 in this class?  It looks like it wouldn't be logged by StandardCmd anymore if these didn't match.

Reply:
Yeah, that is intentional.  If they don't match addr and cmd1 they are probably not meant for this handler.  Maybe for another device? or another message to this device?

If they make it to the end of the handler list without matching anything, we will print a different error message.